.go-version

```diff
@@ -1 +1 @@
-1.22.6
+1.23.4
```

Comment:
Should we update the go version in [this go.mod](https://github.com/elastic/package-spec/blob/7eb29caa329d3adb3eeedfab756e716a63ce948d/go.mod#L3) ? @jsoriano
Or add the toolchain too as in the compliance go.mod?